src/gwc/src/main/java/org/geoserver/gwc/layer/GeoServerTileLayer.java

```diff
@@ -752,6 +755,27 @@ private Map<String, String> buildGetMap(final ConveyorTile tile, final MetaTile
         params.put("TRANSPARENT", "true");
         params.put(GWC_SEED_INTERCEPT_TOKEN, "true");
 
+        // we have the layer's ID (the one with a GUID), use the catalog to get its workspace name
+        GeoServer gs = GeoServerExtensions.bean(GeoServer.class);
+        // gs might be null in test case runs.
+        if ((gs != null) && !gs.getGlobal().isGlobalServices()) {
+            PublishedInfo publishedInfo = getPublishedInfo();
+            if (publishedInfo instanceof LayerInfo) {
+                LayerInfo layerInfo = (LayerInfo) publishedInfo;
+                params.put("WORKSPACE", layerInfo.getResource().getNamespace().getName());
+            } else if (publishedInfo instanceof LayerGroupInfo) {
+                LayerGroupInfo groupInfo = (LayerGroupInfo) publishedInfo;
+                WorkspaceInfo workspace = groupInfo.getWorkspace();
+                if (workspace == null) {
+                    throw new ParameterException(
+                            "Global web services are disabled, global LayerGroup "
+                                    + groupInfo.getName()
+                                    + " inaccessible");
+                }
+                params.put("WORKSPACE", workspace.getName());
```

Comment:
I'm not familiar with having a "WORKSPACE" parameter in a GetMap... where is it used?
I was expecting to see it used in LocalWorkspaceCallback, but did not found a usage.

Reply:
Hi,

This param is picked up here:
https://github.com/geoserver/geoserver/blob/main/src/gwc/src/main/java/org/geoserver/gwc/GWC.java#L1340

That is then picked up here;
https://github.com/geoserver/geoserver/blob/main/src/gwc/src/main/java/org/geoserver/gwc/FakeHttpServletRequest.java#L138

And then gets put into a LocalWorkspace

Reply:
(no action)